internal/pkg/cli/deploy/deploy.go

```diff
@@ -1176,23 +1212,32 @@ func validateAppVersion(appName string, appVersionGetter versionGetter) error {
 	return nil
 }
 
-func validateLBWSRuntime(app *config.Application, envName string, mft *manifest.LoadBalancedWebService, appVersionGetter versionGetter) error {
-	if app.Domain == "" && mft.HasAliases() {
-		log.Errorf(aliasUsedWithoutDomainFriendlyText)
-		return errors.New("alias specified when application is not associated with a domain")
+func validateLBWSRuntime(app *config.Application, env *config.Environment, mft *manifest.LoadBalancedWebService, appVersionGetter versionGetter, validator aliasCertValidator) error {
+	if !mft.HasAliases() {
+		return nil
+	}
+	if env.HasImportedCerts() {
+		aliases, err := mft.RoutingRule.Alias.ToStringSlice()
+		if err != nil {
+			return fmt.Errorf("convert aliases to string slice: %w", err)
+		}
+		if err := validator.ValidCertificatesAliases(aliases, env.CustomConfig.ImportCertARNs); err != nil {
+			return fmt.Errorf("validate aliases against the imported certificate for env %s: %w", env.Name, err)
+		}
+		return validateLBSvcAlias(mft.NLBConfig.Aliases, app, env.Name)
```

Comment:
If cert is imported but DNS isn't delegated, I think NLB alias doesn't work : (

Would it be clearer if we write it like this:

1. Validate the NLB aliases.
1.1 NLB aliases used only when domain delegation happens.
1.2 NLB aliases has the supported format
2. Validate the ALB aliases.
2.1 If import cert, validate ALB aliases against the cert
2.2 O/W
2.2.1 Validate that domain delegation happens
2.2.2  ALB alias has the supported format

Reply:
> If cert is imported but DNS isn't delegated, I think NLB alias doesn't work : (

Good call!

> super nit: alternatively we can do validateLBSvcAliasright after the if !mft.HasAliases() block so that it doesn't have to be repeated at the end of two if blocks.

I still think the current validation ordering makes more sense to me. For example, if their app isn't associated with any domain and they don't import cert at all, it doesn't make sense for us to check if their aliases are valid against the hosted zone (since they don't exist at all)